include/libdnf5/base/log_event.hpp

```diff
@@ -41,30 +41,32 @@ class LogEvent {
         libdnf5::GoalProblem problem,
         const std::set<std::string> & additional_data,
         const libdnf5::GoalJobSettings & settings,
-        const libdnf5::transaction::TransactionItemType spec_type,
+        libdnf5::transaction::TransactionItemType spec_type,
         const std::string & spec);
     LogEvent(libdnf5::GoalProblem problem, const SolverProblems & solver_problems);
-    ~LogEvent() = default;
+    ~LogEvent();
+
+    LogEvent(const LogEvent & src);
+    LogEvent & operator=(const LogEvent & src);
+
+    LogEvent(LogEvent && src) noexcept;
+    LogEvent & operator=(LogEvent && src) noexcept;
 
     /// @return GoalAction for which goal event was created
-    libdnf5::GoalAction get_action() const { return action; };
+    libdnf5::GoalAction get_action() const;
     /// @return GoalProblem that specify the type of report
-    libdnf5::GoalProblem get_problem() const { return problem; };
+    libdnf5::GoalProblem get_problem() const;
     /// @return Additional information (internal), that are required for formatted string
-    const std::set<std::string> get_additional_data() const { return additional_data; };
+    std::set<std::string> get_additional_data() const;
```

Comment:
Is the removing of `const` needed or wanted?

Reply:
It is only wanted.
That is clang-tidy complained [readability-const-return-type](https://clang.llvm.org/extra/clang-tidy/checks/readability/const-return-type.html).

Reply:
Oh I maybe see now the reason. It is returned by value, maybe we can change it to `const &` ?

Reply:
I like that, updated.